pkg/checkmarx/checkmarx.go

```diff
@@ -0,0 +1,469 @@
+}
+
+// Scan - Scan Structure
+type Scan struct {
+	ID   int  `json:"id"`
+	Link Link `json:"link"`
+}
+
+// ResultsStatistics - ResultsStatistics Structure
+type ResultsStatistics struct {
+	High   int `json:"highSeverity"`
+	Medium int `json:"mediumSeverity"`
+	Low    int `json:"lowSeverity"`
+	Info   int `json:"infoSeverity"`
+}
+
+// ScanStatus - ScanStatus Structure
+type ScanStatus struct {
+	ID       int    `json:"id"`
+	Link     Link   `json:"link"`
+	Status   Status `json:"status"`
+	ScanType string `json:"scanType"`
+	Comment  string `json:"comment"`
+}
+
+// Status - Status Structure
+type Status struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+}
+
+// Project - Project Structure
+type Project struct {
+	ID                 int                `json:"id"`
+	TeamID             string             `json:"teamId"`
+	Name               string             `json:"name"`
+	IsPublic           bool               `json:"isPublic"`
+	SourceSettingsLink SourceSettingsLink `json:"sourceSettingsLink"`
+	Link               Link               `json:"link"`
+}
+
+// Team - Team Structure
+type Team struct {
+	ID       string `json:"id"`
+	FullName string `json:"fullName"`
+}
+
+// Link - Link Structure
+type Link struct {
+	Rel string `json:"rel"`
+	URI string `json:"uri"`
+}
+
+// SourceSettingsLink - SourceSettingsLink Structure
+type SourceSettingsLink struct {
+	Type string `json:"type"`
+	Rel  string `json:"rel"`
+	URI  string `json:"uri"`
+}
+
+// Checkmarx is the client communicating with the Checkmarx backend
+type Checkmarx struct {
+	serverURL string
+	client    piperHttp.Client
+	logger    *logrus.Entry
+}
+
+// NewCheckmarx returns a new Checkmarx client for communicating with the backend
+func NewCheckmarx(serverURL, username, password string ) (*Checkmarx, error) {
+	cmx := Checkmarx {
+		serverURL: serverURL,
+		logger: log.Entry().WithField("package", "SAP/jenkins-library/pkg/checkmarx"),
+		client: piperHttp.Client {},
+	}
+
+	token, err := getOAuth2Token(serverURL, username, password)
+	if err != nil {
+		return &cmx, errors.Wrap(err, "error fetching oAuth token")
+	}
+
+	options := piperHttp.ClientOptions {
+		Token: token,
+	}
+	
+	cmx.client.SetOptions(options)
+	
+	return &cmx, nil
+}
+
+func getOAuth2Token(serverURL, username, password string) (string, error) {
+	resp, err := http.PostForm(serverURL+"auth/identity/connect/token",
+		url.Values{
+			"username":      {username},
+			"password":      {password},
+			"grant_type":    {"password"},
+			"scope":         {"sast_rest_api"},
+			"client_id":     {"resource_owner_client"},
+			"client_secret": {"014DF517-39D1-4453-B7B3-9930C563627C"},
+		})
+	if err != nil {
+		return "", err
+	}
+
+	if resp.StatusCode == http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
+		defer resp.Body.Close()
+
+		var token AuthToken
+		json.Unmarshal(body, &token)
+		return token.TokenType + " " + token.AccessToken, nil
+	}
+	data, _ := ioutil.ReadAll(resp.Body)
+	resp.Body.Close()
```

Comment:
can be solved with `defer resp.Body.Close()` right after the call

This avoids duplicates ...